### PR TITLE
Lidar test

### PR DESCRIPTION
**Pull request by ManiacMoustache**
It works for basic distance retrieval.

### CODE REVIEW

File: components/lidar.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    dist = 0
+
+    MXPserial: SerialPort
+
+    bufferArray = bytearray(18)
+
+    def on_enable(self):
+        self.read()
+
+    def checkAndRead(self, packet):
```
Comment (ChrisHirsch) [inline components/lidar.py:15]: Can you add a brief description on what checkAndRead() does and maybe drop in an example packet @Leaf-life can maybe help with this too.

**Review by Leaf-life — APPROVED** [review]

File: robot.py [hunk 2]
```diff
@@ -1,166 +1,177 @@
...
+    def createObjects(self):
+        """
+        Robot-wide initialization code should go here. Replaces robotInit
+        """
+        self.map = RobotMap()
+        self.xboxMap = XboxMap(XboxController(1), XboxController(0))
+
+        self.MXPserial = SerialPort(115200, SerialPort.Port.kMXP, 8,
+        SerialPort.Parity.kParity_None, SerialPort.StopBits.kStopBits_One)
+        self.MXPserial.setReadBufferSize(18)
```
Comment (ChrisHirsch) [inline robot.py]: Hardcoded value. Make it a variable so we can eventually change in config or at least document it and make it easily changed.

**Review by ChrisHirsch — COMMENTED** [review]
Lets get these two changes in and we can merge it.

**Review by ChrisHirsch — APPROVED** [review]
We need to get the CR/LF vs LF fixed somewhere so we don't have all these whitespace changes

**Pull request merged**